pymc_marketing/mlflow.py

```diff
@@ -0,0 +1,211 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+import json
+import os
+from functools import wraps
+from pathlib import Path
+
+import arviz as az
+import pymc as pm
+from pymc.model.core import Model
+
+try:
+    import mlflow
+except ImportError:
+    msg = "This module requires mlflow. Install using `pip install mlflow`"
+    raise ImportError(msg)
+
+from mlflow.utils.autologging_utils import autologging_integration
+
+from pymc_marketing.mmm import MMM
+
+FLAVOR_NAME = "pymc"
+
+
+def save_arviz_summary(idata: az.InferenceData, path: str | Path, var_names) -> None:
+    df_summary = az.summary(idata, var_names=var_names)
+    df_summary.to_html(path)
+    mlflow.log_artifact(str(path))
+    os.remove(path)
+
+
+def save_data(model: Model, idata: az.InferenceData) -> None:
+    features = {
+        var.name: idata.constant_data[var.name].to_numpy()
+        for var in model.data_vars
+        if var.name in idata.constant_data
+    }
+    targets = {
+        var.name: idata.observed_data[var.name].to_numpy()
+        for var in model.observed_RVs
+        if var.name in idata.observed_data
+    }
+
+    data = mlflow.data.from_numpy(features=features, targets=targets)
+    mlflow.log_input(data, context="sample")
+
+
+def save_model_graph(model: Model, path: str | Path) -> None:
+    try:
+        graph = pm.model_to_graphviz(model)
+    except ImportError:
+        return None
```

Comment:
Shall we raise here something more informative so that this does not pass quietly?

Reply:
I have a test for the behavior. If the import doesn't work then it won't be logged via MLflow. 
Should it be silent? I don't think it should raise

Reply:
I think if the import fails and the user actively wants to logs and something fails then we should at least get a warning

Reply:
Added a logging info level. That feel sufficient?